client/src/modules/events/pages/eventPage.tsx

```diff
@@ -149,37 +146,37 @@ export const EventPage: NextPage = () => {
           title: 'Join this event?',
           body: `Note: joining this event will make you a member of the event's chapter.`,
         };
-    const ok = await confirm(confirmOptions);
 
-    if (ok) {
-      if (user) {
+    const loggedOutConfirmOptions = {
+      title: 'Would you like to log in and join this event?',
+      body: `Note: joining this event will make you a member of the event's chapter.`,
+    };
+
+    if (user) {
+      const ok = await confirm(loggedInConfirmOptions);
+      if (ok) {
         await onRsvp(rsvpStatus);
         return;
       }
-      modalProps.onOpen();
-      const {
-        data: { me },
-      } = await login();
-      modalProps.onClose();
-      const eventUser = data?.event?.event_users.find(
-        ({ user: event_user }) => event_user.id === me?.id,
-      );
-      await onRsvp(eventUser?.rsvp.name);
+    } else if (!user) {
+      const ok = await confirm(loggedOutConfirmOptions);
+      if (ok) {
+        modalProps.onOpen();
+        const {
+          data: { me },
+        } = await login();
+        modalProps.onClose();
+        const eventUser = data?.event?.event_users.find(
+          ({ user: event_user }) => event_user.id === me?.id,
+        );
+        await onRsvp(eventUser?.rsvp.name);
+      }
     }
   }
 
-  // TODO: reimplment this the login modal with Auth0
-  async function checkOnCancelRsvp() {
-    await onCancelRsvp();
-  }
-
   useEffect(() => {
     if (canShowConfirmationModal && !hasShownModal) {
-      if (!rsvpStatus || rsvpStatus === 'no') {
-        checkOnRsvp({ invited: true });
-      } else {
-        checkOnCancelRsvp();
-      }
+      tryToRsvp();
```

Comment:
```suggestion
      tryToRsvp({ invited: true });
```
This is reason why one of the tests fails

Reply:
Got it thanks!